Fixes issue #86

KikkiZ opened the pull request:
The document has been updated with a few additional changes to the links in the document.

In the process of updating, I accidentally synchronized my updates to the code to the repo, I have tried to roll back, not sure if it really solved the error.

PS: This update looks like a huge one, the documentation has changed a lot

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+* 创建/清空 `build/`
+* 复制 `addon/**` 到 `build/addon/**`
+* 替换占位符：使用 `replace-in-file` 去替换在 `package.json` 中定义的关键字和配置 (`xhtml`、`.flt` 等)
+* 准备语言文件到 [avaid conflict](https://www.zotero.org/support/dev/zotero_7_for_developers#avoiding_localization_conflicts)
```
```suggestion
* 准备本地化文件以 [避免冲突](https://www.zotero.org/support/dev/zotero_7_for_developers#avoiding_localization_conflicts)
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+> [!note]
+> 在此模板中，release-it 被配置为在本地升级版本、构建、推送提交和 git 标签，随后GitHub Action 将重新构建插件并将 XPI 发布到 GitHub Release. 
+>
+> 如果你需要发布一个本地构建的 XPI，设置  `package.json` 中的 `release-it.github.release` 为 `true` 然后移除 `.github/workflows/release.yml`. 此外，你需要设置环境变量 `GITHUB_TOKEN`，它在[这里](https://github.com/settings/tokens)获取.
```
```suggestion
> 如果你需要发布一个本地构建的 XPI，将 `package.json` 中的 `release-it.github.release` 设置为 `true`，然后移除 `.github/workflows/release.yml`。此外，你需要设置环境变量 `GITHUB_TOKEN`，[获取 GitHub Token](https://github.com/settings/tokens)。
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+
+#### 关于预发布
+
+该模板将 `prerelease` 定义为插件的测试版，当你在 release-it 中选择 `prerelease` 版本 (版本号中带有 `-` )，构建脚本将创建一个新的 `update-beta.json` 给预发布版本使用，这将确保常规版本的用户无法更新到测试版，只有手动下载并安装了测试版的用户才能自动更新到下一个测试版. 当下一个常规版本更新时，将更新 `update.json` 和 `update-beta.json`，以便常规版和测试版用户都可以更新到最新的常规版本.
```
```suggestion
该模板将 `prerelease` 定义为插件的测试版，当你在 release-it 中选择 `prerelease` 版本 (版本号中带有 `-` )，构建脚本将创建 `update-beta.json` 供预发布版本使用，这将确保正式版本的用户不会自动更新到测试版，只有手动下载并安装了测试版的用户才能自动更新到下一个测试版。当下一个正式版更新时，脚本将同时更新 `update.json` 和 `update-beta.json`，以便正式版和测试版用户都可以更新到最新的正式版。
```

northword commented on doc/README-zhCN.md:
```diff
@@ -7,7 +7,22 @@
+
+📜 [Zotero 源代码](https://github.com/zotero/zotero)
+
+📌 [Zotero 插件模板](https://github.com/windingwind/zotero-plugin-template) (即当前库)
```
```suggestion
📌 [Zotero 插件模板](https://github.com/windingwind/zotero-plugin-template) (即本仓库)
```

northword commented on doc/README-zhCN.md:
```diff
@@ -7,7 +7,22 @@
+📌 [Zotero 插件模板](https://github.com/windingwind/zotero-plugin-template) (即当前库)
+
+> [!tip]
+> 👁 关注此库，以便在有修复或更新时及时收到通知.
```
```suggestion
> 👁 Watch 本仓库，以及时收到修复或更新的通知.
```

northword commented on doc/README-zhCN.md:
```diff
@@ -50,19 +52,21 @@
+- 集成Prettier和ES Lint;
+
+> [!warning]
+> Zotero系统已升级(dtd 已弃用，我们将不在使用 .properties). 主分支将只支持 Zotero 7.0.0-beta.12 或更高版本. 如果需要支持 Zotero 6，可能需要同时使用`dtd`、`properties` 和`ftl`. 请参考此库的 `zotero6-bootstrap` 分支.
```
```suggestion
> Zotero 本地化系统已升级（`dtd` 已弃用，我们将不在使用 `.properties`)。 主分支目前只支持 Zotero 7.0.0-beta.12 及更高版本。如果需要支持 Zotero 6，你可能需要同时使用 `dtd`、`properties` 和`ftl`， 请参考 `zotero6-bootstrap` 分支。
```

windingwind replied:
不在 -> 不再

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 ## Quick Start Guide 快速入门指南
 
-### 安装预构建 `xpi`
+### 0 环境要求(Requirement)
```
```suggestion
### 0 前置要求
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+### 0 环境要求(Requirement)
 
-通过直接在GitHub中下载构建好的 `xpi` 文件并将其安装到Zotero中来了解示例的工作原理.
+1. 安装beta版Zotero: https://www.zotero.org/support/beta_builds
```
```suggestion
1. 安装测试版 Zotero: https://www.zotero.org/support/beta_builds
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 
-这也是你发布插件的格式，同时这也将是其他人可以直接使用的版本.
+> [!note]
+> 本指南默认你已经对Zotero插件的基础结构和工作原理有初步的了解. 如果你还不了解，请先参考 [文档](https://www.zotero.org/support/dev/zotero_7_for_developers) 和官方插件样例 [Make It Red](https://github.com/zotero/make-it-red) .
```
```suggestion
> 本指南默认你已经对 Zotero 插件的基本结构和工作原理有初步的了解. 如果你还不了解，请先参考 [文档](https://www.zotero.org/support/dev/zotero_7_for_developers) 和官方插件样例 [Make It Red](https://github.com/zotero/make-it-red) 。
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 
-  > 注意设置 addonID 和 addonRef 以避免冲突.
+1. 点击 `Use this template`；
+2. 使用 `git clone` 克隆此库；
```
```suggestion
2. 使用 `git clone` 克隆前一步生成的仓库；
```

clone的是生成的仓库，不是本仓库

northword commented on doc/README-zhCN.md:
````diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
-# release-it: https://github.com/release-it/release-it
-npm run release
-```
+### 2 调整模板设置和环境(Config Template Settings and Enviroment)
````
```suggestion
### 2 配置模板和开发环境
```

northword commented on doc/README-zhCN.md:
````diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
-   cp ./scripts/zotero-cmd-default.json ./scripts/zotero-cmd.json
-   vim ./scripts/zotero-cmd.json
-   ```
+如果你需要在GitHub意外的地方托管你的 XPI 包，请删除 `releasePage` 并添加 `updateLink`，并将值设置为你的 XPI 下载地址.
````
```suggestion
如果你需要在 GitHub 以外的地方托管你的 XPI 包，请删除 `releasePage`，添加 `updateLink`，并将值设置为你的 XPI 下载地址。
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+如果你需要在GitHub意外的地方托管你的 XPI 包，请删除 `releasePage` 并添加 `updateLink`，并将值设置为你的 XPI 下载地址.
 
-4. 构建插件并使用 `npm run restart` 重启 Zotero.
+2. 复制 Zotero 命令行配置文件. 修改启动你之前安装的 beta 版 Zotero 的命令
```
```suggestion
2. 复制 Zotero 启动配置，填入 Zotero 可执行文件路径和 profile 路径
```

windingwind commented on README.md:
```diff
@@ -278,14 +278,14 @@ npm run release
 > In this template, release-it is configured to locally bump the version, build, and push commits and git.tags, subsequently GitHub Action will rebuild the plugin and publish the XPI to GitHub Release.
 >
-> If you need to release a locally built XPI, set `release-it.github.release` to `true` in `package.json` and remove `.github/workflows/release.yml`. Besides that, you need to set the environment variable `GITHUB_TOKEN`, get it in <https://github.com/settings/tokens>
+> If you need to release a locally built XPI, set `release-it.github.release` to `true` in `package.json` and remove `.github/workflows/release.yml`. Besides that, you need to set the environment variable `GITHUB_TOKEN`, get it in [here](https://github.com/settings/tokens).
```
I suppose @northword put these links' actual url here intentionally?

northword replied:
这个倒是不影响，虽然我确实是故意直接放网址的 :>

northword replied:
这个倒是不影响，虽然我确实是故意直接放网址的 :>

KikkiZ replied:
okok，我之后会注意修改链接的内容，尽量避免 `here` 这类标签

northword commented on doc/README-zhCN.md:
````diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 
-6. 在 Firefox 中，转到devtools，转到设置，单击 "enable remote debugging" ，同时，旁边的按钮也是关于调试的。
+	```sh
+	cp ./scripts/zotero-cmd-default.json ./scripts/zotero-cmd.json
````
```suggestion
	cp ./scripts/zotero-cmd-template.json ./scripts/zotero-cmd.json
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+3. 运行 `npm install` 以安装相关依赖
 
-7. 在 Zotero 中，进入设置-高级-编辑器，搜索 "debugging" 然后单击 "allow remote debugging".
+	> 如果你使用 `pnpm` 作为你项目的包管理器，你需要添加 `public-hoist-pattern[]=*@types/bluebird*` 到`.npmrc`, 详情请[跳转这里](https://github.com/windingwind/zotero-types?tab=readme-ov-file#usage) .
```
```suggestion
	> 如果你使用 `pnpm` 作为包管理器，你需要添加 `public-hoist-pattern[]=*@types/bluebird*` 到`.npmrc`, 详情请参阅 [zotero-types](https://github.com/windingwind/zotero-types?tab=readme-ov-file#usage) .
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+	> 如果你使用 `pnpm` 作为你项目的包管理器，你需要添加 `public-hoist-pattern[]=*@types/bluebird*` 到`.npmrc`, 详情请[跳转这里](https://github.com/windingwind/zotero-types?tab=readme-ov-file#usage) .
 
-8. 在 Firefox 中连接 Zotero. 在 FireFox 102中，在远程调试页面底部输入 `localhost:6100` 然后单击 `add`.
+### 3 编码(Coding)
```
```suggestion
### 3 开始开发
```

windingwind commented on doc/README-zhCN.md:
```diff
@@ -272,7 +305,7 @@ npm run release
-> 可以在 [`src/index.ts`](https://github.com/windingwind/zotero-plugin-template/blob/bootstrap/src/index.ts) 中查看更多
+> 可以在 [`src/index.ts`](https://github.com/windingwind/zotero-plugin-template/blob/main/src/index.ts)中查看更多
 
 引导插件在沙盒中运行，但沙盒中没有默认的全局变量，例如 `Zotero` 或 `window` 等我们曾在覆盖插件环境中使用的变量.
```
just noticed, `引导插件` should be `bootstrap插件` and `覆盖插件` be `overlay插件`

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 
-10. 点击 "Inspect Main Process"
+* 在开发模式下预构建插件
+* 让 Zotero 从`build/` 中加载插件
```
```suggestion
* 启动 Zotero，令其从 `build/` 中加载插件
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
-10. 点击 "Inspect Main Process"
+* 在开发模式下预构建插件
+* 让 Zotero 从`build/` 中加载插件
+* 开启开发工具（devtool） 
```
```suggestion
* 打开开发者工具（devtool） 
```

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+* 在开发模式下预构建插件
+* 让 Zotero 从`build/` 中加载插件
+* 开启开发工具（devtool） 
+* 注意 `src/**` 和 `addon/**`.
```
```suggestion
* 监听 `src/**` 和 `addon/**`.
```

KikkiZ replied:
我建议将原文中的 `watch` 改为 `listening` 可能会更加易于理解

northword replied:
> 我建议将原文中的 `watch` 改为 `listening` 可能会更加易于理解

watch 和 listening 是两个术语，这里是 watch 没有问题。中文也可以选择翻译为 “监视” 以示区别。

northword commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
+
+### 4 构建(Build)
+
+运行 `npm run build` 在生产模式下构建插件，然后用于安装的 xpi 和构建的代码在 `build` 文件夹中.
```
```suggestion
运行 `npm run build` 在生产模式下构建插件，用于安装的 xpi 和构建的代码在 `build` 文件夹中.
```

northword commented:
> In the process of updating, I accidentally synchronized my updates to the code to the repo, I have tried to roll back, not sure if it really solved the error.

这个包含以往的提交，是因为上一次 PR 是以 squash 的方式合并进仓库的，这意味着 PR 发起（from）分支的所有提交将被压缩为一个提交进入本仓库 main 分支；而 sync to upstream 是创建一个 merge 提交到 fork 仓库，这使得 upstream/main 与 fork/main 无法保持一致。

当向其他仓库 PR 时，最好在新的分支中（从upstream/main 创建分支）提交内容然后发起 PR。

现在的 main 可以在本地强制推送以和本仓库的提交记录保持一致。

windingwind commented on README.md:
```diff
@@ -56,7 +56,7 @@ If you are using this repo, I recommended that you put the following badge on yo
 - Event-driven, functional programming, under extensive skeleton;
 - Simple and user-friendly, works out-of-the-box.
-- ⭐ [New!] Auto hot reload! Whenever the source code is modified, automatically compile and reload. [See here→](#auto-hot-reload)
+- ⭐ [New!] Auto hot reload! Whenever the source code is modified, automatically compile and reload. See more in [section auto hot reload](#auto-hot-reload)
```
[section auto hot reload] seems not to be a meaningful sentence

windingwind commented on README.md:
```diff
@@ -330,7 +330,7 @@ createElement(document, "button", { namespace: "xul" }); // manually set namespa
 ### About Zotero API
 
-Zotero docs are outdated and incomplete. Clone <https://github.com/zotero/zotero> and search the keyword globally.
+Zotero docs are outdated and incomplete. Clone [Zotero Code Repo](https://github.com/zotero/zotero) and search the keyword globally.
```
we'd better keep the raw link here so that people can directly copy the link and clone

KikkiZ replied:
Ok, I'll roll back these changes and take care of these issues later.

windingwind commented on doc/README-zhCN.md:
```diff
@@ -129,136 +133,166 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
-2. 复制 `reload` 、`watch` 和 `start-watch` 命令 `package.json`
-3. 运行 `npm install --save-dev chokidar-cli`
+1. 复制 `scripts/**.mjs`
+2. 复制 `server` 、`build` 和 `stop` 命令 `package.json`
```
复制...到...

windingwind commented on src/modules/examples.ts:
```diff
@@ -191,9 +191,8 @@ export class UIExampleFactory {
-      .getElementById("zotero-item-pane-content")
-      ?.classList.add("makeItRed");
+    document.getElementById("zotero-item-pane-content")
+      ?.classList.add("make");
```
please revert these changes

windingwind replied:
I mean, the changes to the `examples.ts` should be reverted.

KikkiZ replied:
I previously rolled back the code, and I don't know why I synchronized the wrong code again. I am really sorry.

KikkiZ commented:
I rolled back all the links in the English document to their original state, and then I only modified a few punctuation marks in the document, and added a git homepage link.
I also changed the hidden links in the Chinese document to be directly displayed, but I still added necessary descriptions before the links so that users can understand the function of the links. If you think this supplement is unnecessary, I can change it back to its original appearance.

windingwind commented:
thanks!
